### PR TITLE
[ci-visibility] Test skipping for `jest`

### PR DESCRIPTION
**Pull request by juan-fernandez**
**Important**: do not merge yet 

### What does this PR do?
Fetch skippable suites (test files) from the API and skip them.

### Motivation
Test skipping part from the intelligent test runner.

### Plugin Checklist

- [x] Unit tests.

### CODE REVIEW

File: packages/datadog-instrumentations/src/jest.js [hunk 1]
```diff
@@ -179,17 +183,42 @@ addHook({
...
-      return result
+
+    try {
+      skippableSuites = await skippableSuitesPromise
+    } catch (e) {
+      // ignore errors
+    }
+    const isTestsSkipped = !!skippableSuites.length
+
+    const result = await runCLI.apply(this, arguments)
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/jest.js:206]: this is where `jest` tests run

File: packages/datadog-instrumentations/src/jest.js [hunk 2]
```diff
@@ -249,6 +278,12 @@ addHook({
 
 function configureTestEnvironment (readConfigsResult) {
   const { configs } = readConfigsResult
+  configs.forEach(config => {
+    skippableSuites.forEach((suite) => {
```
Comment (juan-fernandez) [inline packages/datadog-instrumentations/src/jest.js:282]: the skippable suites that we fetched earlier are set to be ignored here

File: packages/dd-trace/src/encode/coverage-ci-visibility.js [hunk 3]
```diff
@@ -12,7 +12,6 @@ const MAXIMUM_NUM_COVERAGE_FILES = 100
 class CoverageCIVisibilityEncoder extends AgentEncoder {
   constructor () {
     super(...arguments)
-    this.codeCoverageBuffers = []
```
Comment (juan-fernandez) [inline packages/dd-trace/src/encode/coverage-ci-visibility.js:15]: unrelated fix

File: packages/dd-trace/src/proxy.js [hunk 4]
```diff
@@ -65,6 +68,7 @@ class Tracer extends NoopProxy {
           } else {
             log.debug('Successfully uploaded git metadata')
           }
+          gitMetadataUploadFinishCh.publish(err)
```
Comment (juan-fernandez) [inline packages/dd-trace/src/proxy.js:71]: this seemed like a good use of `diagnostics_channel`. If we think it isn't, there are other alternatives for this

File: packages/datadog-plugin-jest/src/index.js [hunk 5]
```diff
@@ -63,6 +67,55 @@ class JestPlugin extends Plugin {
...
+      })
+    })
+
+    this.addSub('ci:jest:test-suite:skippable', ({ onResponse, onError }) => {
+      if (!this.config.isAgentlessEnabled || !this.config.isIntelligentTestRunnerEnabled) {
+        onResponse([])
+        return
+      }
+      // we only request after git upload has happened
+      gitMetadataPromise.then(() => {
```
Comment (juan-fernandez) [inline packages/datadog-plugin-jest/src/index.js:87]: it's important that we fetch the skippable suites _after_ git metadata has been uploaded, otherwise the algorithm does not work

**Review by rochdev — APPROVED** [review]
